.github/CHANGELOG.md

```diff
@@ -83,6 +57,245 @@
    2: ──H──╰C────────RZ(0.4)──RZ(-0.4)──RX(0.3)──╰CY──Y──┤
   ```
 
+* A new quantum function transform has been added to push commuting
+  single-qubit gates through controlled operations.
+  [(#1464)](https://github.com/PennyLaneAI/pennylane/pull/1464)
+
+  The `commute_controlled` transform works as follows:
+
+  ```python
+  def circuit(theta):
+      qml.PauliX(wires=2)
+      qml.S(wires=0)
+      qml.CNOT(wires=[0, 1])
+      qml.PhaseShift(theta/2, wires=0)
+      qml.T(wires=0)
+      qml.Toffoli(wires=[0, 1, 2])
+      return qml.expval(qml.PauliZ(0))
+  ```
+
+  ```pycon
+  >>> optimized_circuit = qml.transforms.commute_controlled(direction="right")(circuit)
+  >>> dev = qml.device('default.qubit', wires=3)
+  >>> qnode = qml.QNode(optimized_circuit, dev)
+  >>> print(qml.draw(qnode)(0.5))
+   0: ──╭C──╭C──S──Rϕ(0.25)──T──┤ ⟨Z⟩
+   1: ──╰X──├C──────────────────┤
+   2: ──────╰X──X───────────────┤
+  ```
```

Comment:
For the compilation transforms, how do we feel about using decorator syntax instead?

```python
dev = qml.device('default.qubit', wires=3)

@qml.qnode(dev)
@qml.transforms.commute_controlled(direction="right")
def circuit(theta):
    qml.PauliX(wires=2)
    qml.S(wires=0)
    qml.CNOT(wires=[0, 1])
    qml.PhaseShift(theta/2, wires=0)
    qml.T(wires=0)
    qml.Toffoli(wires=[0, 1, 2])
    return qml.expval(qml.PauliZ(0))

>>> print(qml.draw(circuit)(0.5))
 0: ──╭C──╭C──S──Rϕ(0.25)──T──┤ ⟨Z⟩
 1: ──╰X──├C──────────────────┤
 2: ──────╰X──X───────────────┤
```

To me, it makes the examples easier to read, and slightly shorter (there is less duplication).

Reply:
:+1:

Reply:
With the condensed bullet list of compilation transforms this example has been removed.

Reply:
@antalszava there is the compilation example on lines 15-43 that can be updated to use decorator syntax :)